File: models/Overflow.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

// Overflow model
type Overflow struct {
	ID       primitive.ObjectID `bson:"_id,omitempty"`
	Title    string             `bson:"title"`
	Content  string             `bson:"content"`
	Username string             `bson:"username"`
	Comments []Comment          `bson:"comments"`
}

File: models/Overflow.js
const mongoose = require('mongoose');

const commentSchema = new mongoose.Schema({
  username: { type: String, required: true },
  text: { type: String, required: true },
  // createdAt: { type: Date, default: Date.now },
});

const overflowSchema = new mongoose.Schema({
  title: { type: String, required: true },
  content: { type: String, required: true },
  author: { type: String, required: true },
  createdAt: { type: Date, default: Date.now },
  comments: [commentSchema],
});

const Overflow = mongoose.model('Overflow', overflowSchema, 'Overflow');

module.exports = Overflow;
